misk-hibernate/src/test/kotlin/misk/hibernate/ReflectionQueryFactoryTest.kt

```diff
@@ -910,6 +910,29 @@ class ReflectionQueryFactoryTest {
     assertThat(jurassicCount).isEqualTo(2)
   }
 
+  @Test
+  fun selectWithIndexHints() {
+    val movieId = transacter.allowCowrites().transaction { session ->
+      val jp = DbMovie("Jurassic Park", LocalDate.of(1993, 6, 9))
+      session.save(jp)
+      val jg = DbActor("Jeff Goldblum")
+      session.save(jg)
+      session.save(DbCharacter("Ian Malcolm", jp, jg))
+
+      jp.id
+    }
+
+    val selectResult = transacter.readOnly().transaction { session ->
```

Comment:
is there a way to test that the hint was actually applied?

Reply:
I wish – I ran into issues with MySQL/Vitess/TiDB treating the inputs slightly differently.
One thing I had in mind was using `IGNORE INDEX` to force a failure when table scans are not allowed on the query. WDYT?

Reply:
Yeah, I think that could be a good test. How are table scans not allowed though?